sammba/modality_processors/perfusion_fair.py

```diff
@@ -0,0 +1,369 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sunday 4th March 2018
+
+@author: Nachiket Nadkarni
+"""
+
+
+import pandas as pd
+import nibabel as nib
+import numpy as np
+from scipy.optimize import least_squares as ls
+from multiprocessing import Pool
+from functools import partial
+import tqdm
+
+
+def perf_fair_read_ptbl(perf_fair_fname):
+    """
+    Extract important information about the perfusion FAIR acquisition
+    parameters from the *_ptbl.txt file created by
+    sammba.io_conversions.dcm_to_nii.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the perfusion *.nii.gz file converted by
+        sammba.io_conversions.dcm_to_nii, NOT to the *_ptbl.txt file itself.
+        The *_ptbl.txt file name is determined automatically fro; that of the
+        *.nii.gz.
+
+    Returns
+    -------
+    Dictionary of parameters useful to perfusion fitting functions.    
+    """
+    
+    # the *_ptbl.txt file will have the same name as the perfusion file itself,
+    # but with _ptbl.txt at the end rather than just .nii.gz. arguably this
+    # function should directly take the text file name as input, but it is

```

Comment:
I prefer keeping things simple. IMHO a function reading a ptbl.txt file should take this file as input. Can't see a reason not to do so

Reply:
It reduces the number of parameters entered further up the chain. In practice, the only function I want the user to touch is perf_fair_niiptbl_proc or maybe perf_fair_nii_proc. I would not be against giving all the other functions leading underscores